components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/internal/CarbonSecurityComponent.java

```diff
@@ -325,137 +327,130 @@ public void onAllRequiredCapabilitiesAvailable() {
-
-                        IdentityStoreConnectorFactory identityStoreConnectorFactory = CarbonSecurityDataHolder
-                                .getInstance().getIdentityStoreConnectorFactoryMap().get(connectorType);
+            DomainConfig domainConfig = DomainConfigBuilder.getDomainConfig(claimConfig.getMetaClaims());
+            CarbonSecurityDataHolder.getInstance().setDomainConfig(domainConfig);
+        } catch (DomainException e) {
+            log.error(e.getMessage(), e);
+        }
 
-                        if (null != identityStoreConnectorFactory) {
-                            IdentityStoreConnector identityStoreConnector = identityStoreConnectorFactory
-                                    .getConnector();
+        // Set domains to domain manager.
+        DomainManager domainManager = null;
+        try {
+            domainManager = this.createDomainManagerFromConfig(CarbonSecurityDataHolder
+                    .getInstance().getDomainConfig());
+        } catch (DomainException e) {
+            log.error(e.getMessage(), e);
+        }
 
+        AuthorizationStore authorizationStore;
+        CredentialStore credentialStore;
+        IdentityStore identityStore;
+
+        if (storeConfig.isCacheEnabled()) {
+            authorizationStore = new CacheBackedAuthorizationStore(storeConfig
+                    .getAuthorizationStoreCacheConfigMap());
+            identityStore = new CacheBackedIdentityStore(storeConfig
+                    .getIdentityStoreCacheConfigMap());
+        } else {
+            identityStore = new IdentityStoreImpl();
+            authorizationStore = new AuthorizationStoreImpl();
+        }
 
-                            domain.addIdentityStoreConnector(identityStoreConnector);
+        credentialStore = new CredentialStoreImpl();
 
-                            domainMetaClaimMappings.put(identityStoreConnectorId, metaClaimMappings.get
-                                    (identityStoreConnectorId));
-                        } else {
-                            log.warn("ConnectorType " + connectorType + " is not available");
-                        }
-                    } else {
-                        log.warn("IdentityStoreConnector " + identityStoreConnectorId + " was not found");
-                    }
-                }
+        try {
+            credentialStore.init(domainManager, storeConfig.getCredentialConnectorConfigMap());
+            identityStore.init(domainManager, storeConfig.getIdentityConnectorConfigMap());
+            authorizationStore.init(storeConfig.getAuthorizationConnectorConfigMap());
+        } catch (CredentialStoreException | AuthorizationStoreException | IdentityStoreException e) {
+            log.error(e.getMessage(), e);
+        }
 
-                domain.setClaimMappings(domainMetaClaimMappings);
+        // Register the carbon realm service.
+        // TODO: Validate the configuration files for multiple primary attributes.
```

Comment:
There will not be multiple primary attributes. Therefore this TODO should be removed